bob-backend/src/core.rs

```diff
@@ -405,25 +415,27 @@ impl Backend {
         let mut keys_by_operations: HashMap<_, (Vec<_>, Vec<_>)> = HashMap::new();
         for (ind, &key) in keys.iter().enumerate() {
             let operation = self.find_operation(key, options);
-            if let Some(operation) = operation {
-                keys_by_operations
-                    .entry(operation)
-                    .and_modify(|(keys, indexes)| {
-                        keys.push(key);
-                        indexes.push(ind);
-                    })
-                    .or_insert_with(|| (vec![key], vec![ind]));
+            if let Some(operations) = operation {
+                for op in operations {
+                    keys_by_operations
+                        .entry(op)
+                        .and_modify(|(keys, indexes)| {
+                            keys.push(key);
+                            indexes.push(ind);
+                        })
+                        .or_insert_with(|| (vec![key], vec![ind]));
+                }
             }
         }
         keys_by_operations
     }
 
-    fn find_operation(&self, key: BobKey, options: &BobOptions) -> Option<Operation> {
-        let (vdisk_id, path) = self.mapper.get_operation(key);
+    fn find_operation(&self, key: BobKey, options: &BobOptions) -> Option<Vec<Operation>> {
+        let (vdisk_id, paths) = self.mapper.get_operation(key);
```

Comment:
Move this call into `if options.get_normal()` branch, because `paths` are only needed there

Reply:
We do need vdisk_id for alien though